src/win/poll.c

```diff
@@ -204,8 +207,16 @@ static void uv__fast_poll_process_poll_req(uv_loop_t* loop, uv_poll_t* handle,
     }
   }
 
-  if ((handle->events & ~(handle->submitted_events_1 |
+  /* UV_DISCONNECT is handled as a UV_READABLE */
```

Comment:
I don't really understand what you mean by "X is handled as Y". It looks a bit like:

```
/* This presence of this comment justifies the following black magic. */
foo *= ~(bar ^~ foo << -2);
```

Could you please write out what actually is going on? Start by explaining that `UV_DISCONNECT` isn't really an event of it's own, but rather an extra bit of information passed along with an `UV_READABLE` event, and that it's not legal to poll of `UV_DISCONNECT` by itself (*). If it *is* legal to poll of `UV_DISCONNECT` in a standalone fashion, this patch isn't valid.

(*) cc @saghul, @bnoordhuis`